remote/gogs/gogs.go

```diff
@@ -177,6 +178,12 @@ func (c *client) File(u *model.User, r *model.Repo, b *model.Build, f string) ([
 	buildRef := b.Commit
 	if buildRef == "" {
 		buildRef = b.Ref
+
+		// Remove refs/tags or refs/heads, Gogs needs a short ref
+		refPath := strings.SplitAfterN(b.Ref, "/", 3)
```

Comment:
How about the following code for high performance?

```go
	refPath := strings.TrimPrefix(
		strings.TrimPrefix(
			b.Ref,
			"refs/heads/",
		),
		"refs/tags/",
	)
```

https://github.com/drone/drone/blob/master/remote/bitbucketserver/convert.go#L105-L111